src/Audio/AudioEngine.cs

```diff
@@ -65,8 +65,23 @@ public bool IsDisposed
 		private readonly FAudio.FACTNotificationCallback xactNotificationFunc;
 		private FAudio.FACTNotificationDescription notificationDesc;
 
+		private class IntPtrComparer : IEqualityComparer<IntPtr>
+		{
+			public bool Equals(IntPtr x, IntPtr y)
+			{
+				return x == y;
+			}
+
+			public int GetHashCode(IntPtr obj)
+			{
+				return (int)obj;
```

Comment:
This may need to be ToInt32()... but at the same time, the pointer truncation makes me a bit uncomfortable. Is it even used? If not, we can just do ToInt32 and call it a day, otherwise we may need to use obj.GetHashCode(), goofy as that sounds.

Reply:
Best to delegate to its gethash code.

Honestly, in this particular case, I don't think it uses it.